pkg/sync/isync.go

```diff
@@ -59,3 +57,10 @@ type DataSync struct {
 	Source   string
 	Type
 }
+
+type ProviderConfig struct {
```

Comment:
should we focus on a per-provider configuration rather than a single one ? 🤔  For example, if we now introduce grpc specific configuration, then this get passed to all sync provider types

Reply:
You're right that the field will be passed to all provider types, but it would be a nil reference of the type. IMO I don't see any issue with using a shared data structure - however, if we feel there is a need to introduce one im happy to make these changes. 

Introducing per-sync configuration objects will also add complexity to the passing / parsing of the configurations as startup commands, which is the main argument for using a single shared type (unmarshal directly into a single shared slice, rather than having a []typeXYZ{} slice to accommodate every new data type, or storing all configurations as interfaces and inferring their types on startup)

Reply:
Another option is to map between the 'general' configuration structure and a sync specific one in the NewXYZ(cofiguration) methods, but this option feels inappropriate